src/quickgui/plugin/qgsquickfeatureform.qml

```diff
@@ -0,0 +1,470 @@
+            width: parent.width
+            height: parent.height
+            color: form.style.backgroundColor
+            opacity: form.style.backgroundOpacity
+          }
+
+          ListView {
+            id: content
+            anchors.fill: parent
+            clip: true
+            section.property: "Group"
+            section.labelPositioning: ViewSection.CurrentLabelAtStart | ViewSection.InlineLabels
+            section.delegate: Component {
+              // section header: group box name
+              Rectangle {
+                width: parent.width
+                height: section === "" ? 0 : form.style.group.height
+                color: form.style.group.backgroundColor
+
+                Text {
+                  anchors { horizontalCenter: parent.horizontalCenter; verticalCenter: parent.verticalCenter }
+                  font.bold: true
+                  text: section
+                }
+              }
+            }
+
+            Connections {
+              target: master
+              onReset: content.contentY = 0
+            }
+
+            model: QgsQuick.SubModel {
+              id: contentModel
+              model: form.model
+              rootIndex: form.model.hasTabs ? form.model.index(currentIndex, 0) : undefined
+            }
+
+            delegate: fieldItem
+          }
+        }
+      }
+    }
+  }
+
+  /**
+   * A field editor
+   */
+  Component {
+    id: fieldItem
+
+    Item {
+      id: fieldContainer
+      visible: Type === 'field'
+      height: childrenRect.height
+
+      anchors {
+        left: parent.left
+        right: parent.right
+        leftMargin: 12 * QgsQuick.Utils.dp
+      }
+
+      Label {
+        id: fieldLabel
+
+        text: Name || ''
+        font.bold: true
+        color: ConstraintValid ? form.style.constraint.validColor : form.style.constraint.invalidColor
+      }
+
+      Label {
+        id: constraintDescriptionLabel
+        anchors {
+          left: parent.left
+          right: parent.right
+          top: fieldLabel.bottom
+        }
+
+        text: ConstraintDescription
+        height: ConstraintValid ? 0 : undefined
+        visible: !ConstraintValid
+
+        color: form.style.constraint.descriptionColor
+      }
+
+      Item {
+        id: placeholder
+        height: childrenRect.height
+        anchors { left: parent.left; right: rememberCheckbox.left; top: constraintDescriptionLabel.bottom }
+
+        Loader {
+          id: attributeEditorLoader
+
+          height: childrenRect.height
+          anchors { left: parent.left; right: parent.right }
+
+          enabled: form.state !== "ReadOnly" && !!AttributeEditable
+
+          property var value: AttributeValue
+          property var config: EditorWidgetConfig
+          property var widget: EditorWidget
+          property var field: Field
+          property var constraintValid: ConstraintValid
+          property var homePath: form.project ? form.project.homePath : ""
```

Comment:
Wouldn't it be better to use null/undefined instead of an empty string if the project is not available? I can't think of a situation, where it's desired to silently assume that the homepath is an empty string.